--- a/bootable/recovery/minuitwrp/events.cpp
+++ b/bootable/recovery/minuitwrp/events.cpp
@@ -61,6 +61,9 @@
 #define VIBRATOR_TIMEOUT_FILE	"/sys/class/timed_output/vibrator/enable"
 #define VIBRATOR_TIME_MS    50
 
+#define LEDS_VIBRATOR_BRIGHTNESS_FILE	"/sys/class/leds/vibrator/brightness"
+#define LEDS_VIBRATOR_TRIGGER_FILE	"/sys/class/leds/vibrator/trigger"
+
 #define LEDS_HAPTICS_DURATION_FILE	"/sys/class/leds/vibrator/duration"
 #define LEDS_HAPTICS_ACTIVATE_FILE	"/sys/class/leds/vibrator/activate"
 
@@ -182,11 +185,18 @@
         write_to_file(VIBRATOR_TIMEOUT_FILE, tout);
     }
 #else
-    if (std::ifstream(LEDS_HAPTICS_ACTIVATE_FILE).good()) {
+    if (std::ifstream(LEDS_VIBRATOR_BRIGHTNESS_FILE).good()) {
+        write_to_file(LEDS_VIBRATOR_TRIGGER_FILE, "timer");
+        write_to_file(LEDS_VIBRATOR_BRIGHTNESS_FILE, "200");
+        usleep(timeout_ms * 1000);
+        write_to_file(LEDS_VIBRATOR_BRIGHTNESS_FILE, "0");
+        write_to_file(LEDS_VIBRATOR_TRIGGER_FILE, "none");
+    } else if (std::ifstream(LEDS_HAPTICS_ACTIVATE_FILE).good()) {
         write_to_file(LEDS_HAPTICS_DURATION_FILE, tout);
         write_to_file(LEDS_HAPTICS_ACTIVATE_FILE, "1");
-    } else
+    } else {
         write_to_file(VIBRATOR_TIMEOUT_FILE, tout);
+    }
 #endif
     return 0;
 }
